scanner/src/main/kotlin/scanners/fossid/FossIdScanResults.kt

```diff
@@ -148,28 +148,29 @@ internal fun mapSnippetFindings(
 
             // FossID does not return the hash of the remote artifact. Instead, it returns the MD5 hash of the
             // matched file in the remote artifact as part of the "match_file_id" property.
-            val url = checkNotNull(it.url) {
-                "The URL of snippet ${it.id} must not be null."
+            val url = checkNotNull(snippet.url) {
+                "The URL of snippet ${snippet.id} must not be null."
             }
             val snippetProvenance = ArtifactProvenance(RemoteArtifact(url, Hash.NONE))
-            val purl = it.purl ?: "pkg:${urlToPackageType(url)}/${it.author}/${it.artifact}@${it.version}"
+            val purl =
+                snippet.purl ?: "pkg:${urlToPackageType(url)}/${snippet.author}/${snippet.artifact}@${snippet.version}"
```

Comment:
Nit: For next time, the wrapping should rather be like

    val purl = snippet.purl
        ?: "pkg:${urlToPackageType(url)}/${snippet.author}/${snippet.artifact}@${snippet.version}"

as that also more nicely visualizes default vs. fallback value.